File: leetcode/0066-plus-one/Solution.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        for (int i = digits.length-1; i >= 0; i--) {
            digits[i] = digits[i] + carry;
            carry = digits[i] / 10;
            digits[i] = digits[i] % 10;
        }
        
        if (carry == 0)
            return digits;
        
        int[] a = new int[digits.length+1];
        a[0] = 1;
        for (int i = 1; i < a.length; i++)
            a[i] = digits[i-1];
        
        return a;
    }
}

File: leetcode/0066-plus-one/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in range(len(digits) - 1, -1, -1):
            tmp = digits[i] + carry
            digits[i] = tmp % 10
            carry = tmp // 10
        
        if carry == 0:
            return digits
        
        digits.insert(0, 1)
        
        return digits